### PR TITLE
DOC-746: New Spell Checker Pro API, Commands and Events

### PR DESCRIPTION
**Pull request by JamesToohey**
**Related Ticket:** 

TINY-6416

**Description of Changes:**

Documents new:
- Commands
- API
- Events

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] ~~`_data/nav.yml` has been updated (if applicable)~~
- [x] Files has been included where required (if applicable)
- [x] ~~Files removed have been deleted, not just excluded from the build (if applicable)~~
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: release-notes/release-notes57.md [hunk 1]
```diff
@@ -75,13 +75,34 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **S
 
 **Spell Checker Pro** 2.3.0 introduces enhancements to the `spellchecker_ignore_list` (formerly `spellchecker_whitelist`, see: [The `spellchecker_whitelist` option has been renamed](#thespellchecker_whitelistoptionhasbeenrenamed)).
 
-- It is now possible to specify arrays of words for specific languages to be ignored by Spell Checker Pro plugin using the `spellchecker_ignore_list` option.
-- A new `addIgnoredWords` API allows words to be ignored at runtime by the Spell Checker Pro plugin.
+It is now possible to specify arrays of words for specific languages to be ignored by Spell Checker Pro plugin using the `spellchecker_ignore_list` option, see: [Spell Checker Pro - `spellchecker_ignore_list`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#spellchecker_ignore_list).
```
Comment (EkimChau) [inline release-notes/release-notes57.md:78]: _specify arrays of words_ 

do we provide an example of this on the tinymcespellchecker page?

**Review by EkimChau — APPROVED** [review]

File: release-notes/release-notes57.md (hunk 1 above)
Comment (EkimChau) [reply release-notes/release-notes57.md:78]: yes there is :)

**Review by ashfordneil — APPROVED** [review]

File: _includes/plugin-apis/tinymcespellchecker-apis.md [hunk 2]
```diff
@@ -3,6 +3,8 @@
 | Name | Arguments | Description |
 |------| ------| ----------- |
 | addIgnoredWords | words: `string[]`, lang?: `string` | Add an array of words to the `spellchecker_ignore_list` for a specific language. Add the array of words to all languages by omitting the `lang` parameter. |
+| getLanguage | N/A | Get the current language enabled for spellchecking.
```
Comment (lnewson) [inline _includes/plugin-apis/tinymcespellchecker-apis.md:6]: These probably also need the `{{ site.requires_5_7v }}` note.

File: plugins/premium/tinymcespellchecker.md [hunk 3]
```diff
@@ -343,6 +343,32 @@ tinymce.init({
 });
 ```
 
+### SpellcheckerLanguageChanged event
+
+This event fires when the spellchecking language is changed.
+
+{{ site.requires_5_7v }}
```
Comment (lnewson) [inline plugins/premium/tinymcespellchecker.md:350]: This should be above the text I'm pretty sure, or at least that's how it's done elsewhere :shrug:

**Review by lnewson — APPROVED** [review]

File: _includes/plugin-apis/tinymcespellchecker-apis.md [hunk 4]
```diff
@@ -3,6 +3,8 @@
...
+| setLanguage | language: `string` | Set the current language for spellchecking. This must be a supported language code (case sensitive). For a list of supported languages, see: [Spell Checker Pro - `Supported languages`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#supportedlanguages)
```
Comment (tylerkelly13) [inline _includes/plugin-apis/tinymcespellchecker-apis.md:7]:
```suggestion
| setLanguage | language: `string` | Set the current language for spellchecking. This must be a supported language code (case-sensitive). For a list of supported languages, see: [Spell Checker Pro - `Supported languages`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#supportedlanguages)
```

File: plugins/premium/tinymcespellchecker.md [hunk 5]
```diff
@@ -343,6 +343,32 @@ tinymce.init({
...
+  toolbar: 'spellchecker',
+  init_instance_callback: function (editor) {
+    editor.on('SpellcheckerLanguageChanged', function (e) {
+      console.log('Spelling language changed: ' + e.language + '. Previous language: ' + e.prevLanguage);
+    });
+  }
+});
+```
+
+## Commands
```
Comment (tylerkelly13) [inline plugins/premium/tinymcespellchecker.md:367]: can you please move the commands section to above the events section?

File: release-notes/release-notes57.md (hunk 1 above)
Comment (tylerkelly13) [inline release-notes/release-notes57.md:78]:
```suggestion
It is now possible to specify arrays of words for specific languages to be ignored by the Spell Checker Pro plugin using the `spellchecker_ignore_list` option, see: [Spell Checker Pro - `spellchecker_ignore_list`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#spellchecker_ignore_list).
```

File: release-notes/release-notes57.md [hunk 6]
```diff
@@ -75,13 +75,34 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **S
...
-For information on:
+#### New Spell Checker Pro API
+
+**Spell Checker Pro** 2.3.0 includes a new API with the following methods:
+
+- `addIgnoredWords`
+- `getLanguage`
+- `setLanguage`
+
+For information on the Spell Checker Pro API, see: [Spell Checker Pro - `API`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#apis).
```
Comment (tylerkelly13) [inline release-notes/release-notes57.md:88]:
```suggestion
For information on the Spell Checker Pro API, see: [Spell Checker Pro - APIs]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#apis).
```

File: release-notes/release-notes57.md [hunk 7]
```diff
@@ -75,13 +75,34 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **S
...
+#### New Spell Checker Pro Commands
+
+**Spell Checker Pro** 2.3.0 includes the following new commands:
+
+- `mceSpellcheckEnable`
+- `mceSpellcheckDisable`
+- `mceSpellcheckDialog`
+- `mceSpellcheckDialogClose`
+
+For information on the Spell Checker Pro Commands, see: [Spell Checker Pro - `Commands`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#commands).
```
Comment (tylerkelly13) [inline release-notes/release-notes57.md:99]:
```suggestion
For information on the Spell Checker Pro Commands, see: [Spell Checker Pro - Commands]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#commands).
```

File: release-notes/release-notes57.md [hunk 8]
```diff
@@ -75,13 +75,34 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **S
...
+
+#### New Spell Checker Pro Event
+
+**Spell Checker Pro** introduces a new `SpellcheckerLanguageChanged` which is fired when the active language is changed.
 
-- The `spellchecker_ignore_list` option, see: [Spell Checker Pro - `spellchecker_ignore_list`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#spellchecker_ignore_list).
-- The `addIgnoredWords` API, see: [Spell Checker Pro - `API`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#api).
+For more information on the Spell Checker Pro Events, see: [Spell Checker Pro - `Events`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#events).
```
Comment (tylerkelly13) [inline release-notes/release-notes57.md:105]:
```suggestion
For information on the Spell Checker Pro Events, see: [Spell Checker Pro - `Events`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#events).
```

File: release-notes/release-notes57.md [hunk 9]
```diff
@@ -75,13 +75,34 @@ The {{site.productname}} 5.7 release includes an accompanying release of the **S
...
+- `mceSpellcheckEnable`
+- `mceSpellcheckDisable`
+- `mceSpellcheckDialog`
+- `mceSpellcheckDialogClose`
+
+For information on the Spell Checker Pro Commands, see: [Spell Checker Pro - `Commands`]({{ site.baseurl }}/plugins/premium/tinymcespellchecker/#commands).
+
+#### New Spell Checker Pro Event
+
+**Spell Checker Pro** introduces a new `SpellcheckerLanguageChanged` which is fired when the active language is changed.
```
Comment (tylerkelly13) [inline release-notes/release-notes57.md:103]:
```suggestion
**Spell Checker Pro** introduces a new `SpellcheckerLanguageChanged` event which is fired when the active language is changed.
```

**Review by tylerkelly13 — APPROVED** [review]
Looks good, just a few minor changes before we merge.

**Pull request merged**